File: 卡码/代码随想录/Go/回溯/03-子集/LeetCode_0078.go
//78. 子集
package main

func subsets(nums []int) [][]int {
	res:=make([][]int,0)
	path:=make([]int,0)
	var backtracking func(nums []int,index int)
	backtracking=func(nums []int,index int){
		tmp:=make([]int,len(path))
		copy(tmp,path)
		res=append(res,tmp)
		if index>=len(nums) {
			return
		}
		for i:=index;i<len(nums);i++ {
			path=append(path,nums[i])
			backtracking(nums,i+1)
			path=path[:len(path)-1]
		}
	}
	backtracking(nums,0)
	return res
}

File: 卡码/代码随想录/Python/回溯/03-子集/LeetCode_0078.py
#78. 子集
from typing import *

class Solution:
  def __init__(self):
    self.path:List[int]=[]
    self.res:List[List[int]]=[]
  def backtracking(self,nums:List[int],index:int):
    self.res.append(self.path[:])
    if index>len(nums):
      return
    for i in range(index,len(nums)):
      self.path.append(nums[i])
      self.backtracking(nums,i+1)
      self.path.pop()
  def subsets(self, nums: List[int]) -> List[List[int]]:
    self.backtracking(nums,0)
    return self.res